process/block/baseProcess.go

```diff
@@ -605,10 +605,13 @@ func (bp *baseProcessor) requestHeaderByShardAndNonce(targetShardID uint32, nonc
 	}
 }
 
-func (bp *baseProcessor) cleanupPools(
-	headerHandler data.HeaderHandler,
-	headersPool dataRetriever.HeadersPool,
-) {
+func (bp *baseProcessor) cleanupPools(headerHandler data.HeaderHandler) {
+	headersPool := bp.dataPool.Headers()
+	if headersPool == nil {
```

Comment:
check.ifNil. but this check is done at constructor. no need to do it here.

Reply:
Removed